docs/program/extend/modular-resources/_index.md

```diff
@@ -1,150 +1,528 @@
 
-Viam represents models with *colon-delimited-triplets*.
-For example, the __rdk:builtin:gpio__ model of the __rdk:component:motor__ API provides RDK support for GPIO-controlled DC motors and the __rdk:builtin:DMC4000__ model of the same __rdk:component:motor__ API provides support for the DMC 4000 motor.
+Models are also uniquely namespaced as colon-delimited-triplets in the form of `namespace:family:name`.
+
+For example:
+
+- The `rdk:builtin:gpio` model of the `rdk:component:motor` API provides RDK support for [GPIO-controlled DC motors](/components/motor/gpio/).
+- The `rdk:builtin:DMC4000` model of the same `rdk:component:motor` API provides RDK support for the [DMC4000](/components/motor/dmc4000/) motor.
 
 A common use-case for modular resources is to create a new model using an existing Viam API.
 However, you can also create and expose new API types using modular resources.
 
-## Modular Resource Management with the RDK
+## Use a modular resource with your robot
 
-### Dependency Management
+Add a modular resource to your robot configuration in five steps:
 
-Modular resources may depend on other built-in resources or other modular resources, and vice versa.
-The Viam RDK handles dependency management.
+1. Code a module in Go or Python that implements a new resource and registers the component in the Viam RDK's [global registry of robotic parts](https://github.com/viamrobotics/rdk/blob/main/registry/registry.go).
+2. Create an executable file that runs your module.
+3. Save the executable in a location your `viam-server` instance can access.
+4. Add a **module** referencing this executable to the configuration of your robot.
+5. Add a new component or service referencing the custom resource provided by the configured **module** to the configuration of your robot.
 
-### Startup
+{{% alert title="Modules vs. modular resources" color="tip" %}}
 
-RDK ensures that any configured modules are loaded automatically on startup, and that configured modular resource instances are started alongside configured built-in resources instances.
+A configured **module** can make one or more *modular resources* available for configuration.
 
-### Reconfiguration
+{{% /alert %}}
 
-When you reconfigure a Viam robot (meaning, when you change its configuration), the behavior of modular resource instances versus built-in resource instances is equivalent.
-This means you can add, modify, and remove a modular resource instance from a running robot as normal.
+### Code your module
 
-### Shutdown
+Code a module in the Go or Python programming languages with [Viam's SDKs](/program/sdk-as-client) that does the following:
 
-During robot shutdown, the RDK handles modular resource instances similarly to built-in resource instances - it signals them for shutdown in topological (dependency) order.
+1. Implements a new resource, including implementing any methods the Viam RDK defines for the API of a built-in type if defining a new model (ex. `rdk:component:base`)
```

Comment:
I would expand this to multiple steps that actually explain what's needed. Otherwise it feels like "How to draw an owl" ( https://seths.blog/2014/01/how-to-draw-an-owl/ )

Tentative breakdown might look like:

1. If creating a new API, define its messages and methods in protobuf, then generate code in your chosen language (Go or Python) and use the generated code to implement the higher level server and client functions required.

2. Create a new resource model, making sure to implement all required methods defined in the chosen API (either your own, or a pre-existing API.)

3. Create a main program that will serve as the module itself, using the module helpers provided by your chosen SDK

4. Import both the required API(s) and model(s) into the main program, and register them with the module helper SDK.

5. Compile and/or package your program.